services/construction.go

```diff
@@ -124,3 +145,713 @@ func (s *constructionAPIService) ConstructionSubmit(
+			loggerCons.Error("ConstructionPayloads: add escrow from doesn't match signer",
+				"from", request.Operations[1].Account.Address,
+				"signer", signWithAddr,
+			)
+			return nil, ErrMalformedValue
+		}
+		amount, err := readOasisCurrencyNeg(request.Operations[1].Amount)
+		if err != nil {
+			loggerCons.Error("ConstructionPayloads: add escrow from amount",
+				"amount", request.Operations[1].Amount,
+				"err", err,
+			)
+			return nil, ErrMalformedValue
+		}
+
+		var escrowAccount staking.Address
+		if err = escrowAccount.UnmarshalText([]byte(request.Operations[2].Account.Address)); err != nil {
+			loggerCons.Error("ConstructionPayloads: add escrow account UnmarshalText",
+				"addr", request.Operations[2].Account.Address,
+				"err", err,
+			)
+		}
+		amount2, err := readOasisCurrency(request.Operations[2].Amount)
+		if err != nil {
+			loggerCons.Error("ConstructionPayloads: add escrow account amount",
+				"amount", request.Operations[2].Amount,
+				"err", err,
+			)
+			return nil, ErrMalformedValue
+		}
+		if amount.Cmp(amount2) != 0 {
+			loggerCons.Error("ConstructionPayloads: add escrow amounts differ",
+				"amount_from", amount,
+				"amount_to", amount2,
+				"err", err,
+			)
+			return nil, ErrMalformedValue
+		}
+
+		body = cbor.Marshal(staking.Escrow{
+			Account: escrowAccount,
+			Tokens:  *amount,
+		})
+	} else if len(request.Operations) == 2 &&
+		request.Operations[1].Type == OpTransfer &&
+		request.Operations[1].Account.SubAccount != nil &&
+		request.Operations[1].Account.SubAccount.Address == SubAccountEscrow {
+		loggerCons.Debug("ConstructionPayloads: matched reclaim escrow")
+		method = staking.MethodReclaimEscrow
+
+		var escrowAccount staking.Address
+		if err = escrowAccount.UnmarshalText([]byte(request.Operations[1].Account.Address)); err != nil {
+			loggerCons.Error("ConstructionPayloads: reclaim escrow from UnmarshalText",
+				"addr", request.Operations[1].Account.Address,
+				"err", err,
+			)
+		}
+		amount, err := readPoolShareNeg(request.Operations[1].Amount)
+		if err != nil {
+			loggerCons.Error("ConstructionPayloads: reclaim escrow from amount",
+				"amount", request.Operations[1].Amount,
+				"err", err,
+			)
+			return nil, ErrMalformedValue
+		}
+
+		body = cbor.Marshal(staking.ReclaimEscrow{
+			Account: escrowAccount,
+			Shares:  *amount,
+		})
+	} else {
+		loggerCons.Error("ConstructionPayloads: unmatched operations list",
+			"operations", request.Operations,
+		)
+		return nil, ErrNotImplemented
+	}
+
+	tx := transaction.Transaction{
+		Nonce: nonce,
+		Fee: &transaction.Fee{
+			Amount: *feeAmount,
+			Gas:    DefaultGas,
+		},
+		Method: method,
+		Body:   body,
+	}
+
+	txJSON, err := json.Marshal(tx)
+	if err != nil {
+		loggerCons.Error("ConstructionPayloads: marshal transaction",
+			"tx", tx,
+			"err", err,
+		)
+		return nil, ErrMalformedValue
+	}
+	txCBOR := cbor.Marshal(tx)
+	txHex := hex.EncodeToString(txCBOR)
```

Comment:
Is this only for logging? Why not use Base64?

Reply:
oh yeah, this is only for logging now.

base64 is a lot less used in the Rosetta API, they use more hex

Reply:
moved encoding to error branch